### PR TITLE
[misc] Change IsCategoryEnabled to inline function

### PR DESCRIPTION
**Pull request by gjc13**
Rename to avoid conflict with other code bases.

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Review by gerickson — DISMISSED** [review]
Is there any reason that this can't be converted to an inline and placed in the CHIP namespace to eliminate the need for preprocessor decoration in this fashion?

**Review by woody-apple — CHANGES_REQUESTED** [review]
Echoing @gerickson 's request for name-spacing, now that we're touching this.

**Review by jelderton — APPROVED** [review]

File: src/lib/support/logging/CHIPLogging.h [hunk 1]
```diff
@@ -327,11 +327,17 @@ static inline void GetModuleName(char * buf, uint8_t module)
 
 extern uint8_t gLogFilter;
 
-#define IsCategoryEnabled(CAT) ((CAT) <= gLogFilter)
+inline bool IsCategoryEnabled(uint8_t CAT)
```
Comment (gerickson) [inline src/lib/support/logging/CHIPLogging.h:330]: The other inlines in this file have a static qualifier. Is there any reason these do not?

**Review by woody-apple — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Comment by andy31415** [comment]
@gerickson ? is the static inline change sufficient or are we asking for more reorganization?

**Comment by andy31415** [comment]
@gjc13  is this really a rename as the title claims? it looks to me like a move from macro to inline function. Maybe it should be constexpr as well.

**Comment by gjc13** [comment]
> @gjc13 is this really a rename as the title claims? it looks to me like a move from macro to inline function. Maybe it should be constexpr as well.

Already renamed. It cannot be constexpr since `gLogFilter` is involved and this can be set at runtime.

**Review by BroderickCarlin — APPROVED** [review]

**Pull request merged**